parser-java/kvs-archived-media-parallel-processing-image-rekognition-sample/src/main/java/com/amazonaws/kinesisvideo/workers/GetMediaArchivedRekognitionWorker.java

```diff
@@ -0,0 +1,143 @@
+                                                           final Regions region,
+                                                           final AmazonKinesisVideo amazonKinesisVideo,
+                                                           final FragmentSelector fragmentSelector,
+                                                           final MkvElementVisitor elementVisitor,
+                                                           final String listFragmentsEndpoint,
+                                                           final String getMediaForFragmentListEndpoint)
+    {
+
+        return new GetMediaArchivedRekognitionWorker(
+                streamName, awsCredentialsProvider, listFragmentsEndpoint, getMediaForFragmentListEndpoint, region, fragmentSelector, elementVisitor);
+    }
+
+    @Override
+    public void run() {
+        try {
+            log.info("Start ListFragment worker on stream {}", streamName);
+
+            /* ---------------------------- LIST FRAGMENTS SECTION ---------------------------- */
+            ListFragmentsRequest listFragmentsRequest = new ListFragmentsRequest()
+                    .withStreamName(streamName).withFragmentSelector(fragmentSelector).withMaxResults(fragmentsPerRequest);
+
+            log.info(listFragmentsRequest.toString());
+
+            ListFragmentsResult listFragmentsResult = amazonKinesisVideoArchivedMediaListFragments.listFragments(listFragmentsRequest);
+
+
+            log.info("List Fragments called on stream {} response {} request ID {}",
+                    streamName,
+                    listFragmentsResult.getSdkHttpMetadata().getHttpStatusCode(),
+                    listFragmentsResult.getSdkResponseMetadata().getRequestId());
+
+            for (Fragment f: listFragmentsResult.getFragments()) {
+                fragmentNumbers.add(f.getFragmentNumber());
+            }
+
+            String nextToken = listFragmentsResult.getNextToken();
+
+            /* If result is truncated, keep making requests until nextToken is empty */
+            while (nextToken != null) {
+                listFragmentsRequest = new ListFragmentsRequest()
+                        .withStreamName(streamName).withNextToken(nextToken);
+                listFragmentsResult = amazonKinesisVideoArchivedMediaListFragments.listFragments(listFragmentsRequest);
+
+                for (Fragment f: listFragmentsResult.getFragments()) {
+                    fragmentNumbers.add(f.getFragmentNumber());
+                }
+                nextToken = listFragmentsResult.getNextToken();
+            }
+
+            Collections.sort(fragmentNumbers);
+
+            /* ------------------------- GET MEDIA SECTION ------------------------- */
+
+            if (fragmentNumbers.size() > 0) {
+                GetMediaForFragmentListRequest getMediaFragmentListRequest = new GetMediaForFragmentListRequest()
+                        .withFragments(fragmentNumbers)
```

Comment:
Since you're sharing the fragment numbers with the other workers, there'll be some duplicated fragmentNumbers in your request.